docs/interpreter_status.md

```diff
@@ -0,0 +1,216 @@
+this op does not have an expander pass, and supporting this in StableHLO is a
+WIP.
+
+`RngOp` and `RngBitGeneratorOp` can be decomposed into MHLO ops, but the
+decomposition introduces a `XlaRngGetAndUpdateStateOp` which is an MHLO specific
+op. Supporting interpretation of these two ops is a WIP.
+
+<!-- markdownlint-disable line-length -->
+The tool to convert remaining ops in this category to StableHLO ops that the
+interpreter supports resides in [hlo_expand_main.cc](https://github.com/openxla/xla/blob/main/xla/tools/hlo_expand_main.cc).
+<!-- markdownlint-enable line-length -->
+
+### Not in HLO
+
+Apart from the specced ops, this category consists of 10 unspecced ops (as they
+plan to be moved out of StableHLO), some of which have existing passes in
+[mhlo](https://github.com/openxla/xla/tree/main/xla/mlir_hlo/mhlo/transforms) to
+convert them to StableHLO equivalent ops. There are three ops the interpreter
+does not support because there are no existing decompositions to StableHLO ops:
+
+* `compute_reshape_shape`
+* `cstr_reshapable`
+* `trace`
+
+`compute_reshape_shape` and `cstr_reshapable` ops are part of the ongoing
+Dynamism work, and they are planned to be removed from StableHLO (see
+[#1668](https://github.com/openxla/stablehlo/issues/1668)).
+
+`trace` op is private to XLA and there no no users in JAX, PyTorch or TensorFlow
+(see [#604](https://github.com/openxla/stablehlo/issues/604)).
+
+<!-- markdownlint-disable line-length -->
+The tool to convert remaining ops in this category to equivalent StableHLO ops
+that the interpreter supports resides in [mlir-hlo-opt.cc](https://github.com/openxla/xla/blob/main/xla/mlir_hlo/tools/mlir-hlo-opt/mlir-hlo-opt.cc).
+<!-- markdownlint-enable line-length -->
+
+### Quantization
+
+There are two specced ops in this category that the interpreter does not support
+at the moment. These ops are part of an ongoing Quantization work. Supporting
+interpretation of these two ops is a WIP:
+
+* `UniformDequantizeOp`
+* `UniformQuantizeOp`
+
+## Build and Run the Reference Interpreter
+
+The interpreter can be built and tested via Bazel or CMake (see
+[README.md](https://github.com/openxla/stablehlo/blob/main/README.md)). To run
+the interpreter, we have a translate tool to interpret StableHLO programs
+written in MLIR.
+
+```bash
+stablehlo-translate --interpret <path/to/program>
+```
+
+## Testing StableHLO Programs
+
+<!-- markdownlint-disable line-length -->
+We use LLVM's [lit](https://llvm.org/docs/CommandGuide/lit.html) tool to run,
+and compare against expected value to test the interpreter (see [stablehlo/tests](https://github.com/openxla/stablehlo/tree/main/stablehlo/tests)
+for sample tests).
+<!-- markdownlint-enable line-length -->
+
+A custom `Check` dialect is used to compare interpreter runtime values with
+expected values. The `Check` ops are embedded into the StableHLO programs which
+is run via callbacks through `stablehlo.custom_call`.
```

Comment:
Lets leave off custom_call, this is a custom solution for when tests need to be serialized which is out of scope of this doc.